### PR TITLE
PSMDB-1754: don't rotate audit if file already exists

### PR DESCRIPTION
**Pull request by plebioda**
Do not rotate the audit log file if the target file with timestamp suffix already exists.

Anything in this description will be included in the commit message. Replace or delete this text before merging. Add links to testing in the comments of the PR.

### CODE REVIEW

File: src/mongo/db/audit/audit.cpp [hunk 1]
```diff
@@ -219,19 +219,47 @@ class FileAuditLog : public WritableAuditLog {
...
+                    return exceptionToStatus();
+                }
+            }();
+
+            if (!targetExists.isOK()) {
+                // If cannot determine whether the target file exists,
+                // report a minor error and skip renaming.
+                if (onMinorError) {
+                    onMinorError({ErrorCodes::FileRenameFailed,
+                                  "Cannot verify whether destination already exists: {}"_format(
```
Comment (ktrushin) [inline src/mongo/db/audit/audit.cpp]: In my opinion, just logging `Cannot verify ...` can left a user confused because she won't understand the consequentces of that error. I would also add something like `Skipping audit log rotation`.

File: src/mongo/db/audit/audit.cpp [hunk 2]
```diff
@@ -219,19 +219,47 @@ class FileAuditLog : public WritableAuditLog {
...
+                }
+            } else if (targetExists.getValue()) {
+                // If the target file already exists,
+                // report a minor error and skip renaming.
                 if (onMinorError) {
-                    onMinorError(
-                        {ErrorCodes::FileRenameFailed,
-                         "Failed to rename {} to {}: {}"_format(_fileName, s, errorMessage(ec))});
+                    onMinorError({ErrorCodes::FileRenameFailed,
+                                  "Target already exists during audit log rotation."
```
Comment (ktrushin) [inline src/mongo/db/audit/audit.cpp]: Same as above

File: jstests/audit/audit_log_rotate_on_start.js [hunk 3]
```diff
@@ -134,6 +148,68 @@ function runTestReopen(fileName, format) {
         "Expected second file content to start with first file content (file reopened in append mode)");
 }
 
+function runTestRenameFileExist(fileName, format) {
```
Comment (ktrushin) [inline jstests/audit/audit_log_rotate_on_start.js:156]: Since the test covers an edge case that is unlikely to happen in a normally working deployment, I think it is worth adding a description of that edge case as a comment for the test.

**Review by ktrushin — CHANGES_REQUESTED** [review]
